api/src/test/java/org/openmrs/module/fhir2/providers/r4/PersonFhirResourceProviderTest.java

```diff
@@ -283,4 +287,62 @@ public void getPatientHistoryByWithWrongId_shouldThrowResourceNotFoundException(
 		assertThat(resourceProvider.getPersonHistoryById(idType).size(), equalTo(0));
 	}
 	
+	@Test
+	public void updatePerson_shouldUpdatePerson() {
+		Person newPerson = person;
+		
+		when(fhirPersonService.update(PERSON_UUID, person)).thenReturn(newPerson);
+		
+		MethodOutcome result = resourceProvider.updatePerson(new IdType().setValue(PERSON_UUID), person);
+		assertThat(result, CoreMatchers.notNullValue());
+		assertThat(result.getResource(), CoreMatchers.equalTo(newPerson));
+	}
+	
+	@Test(expected = InvalidRequestException.class)
+	public void updatePerson_shouldThrowInvalidRequestExceptionForWrongPersonUuid() {
+		when(fhirPersonService.update(WRONG_PERSON_UUID, person)).thenThrow(InvalidRequestException.class);
+		
+		resourceProvider.updatePerson(new IdType().setValue(WRONG_PERSON_UUID), person);
+	}
+	
+	@Test(expected = MethodNotAllowedException.class)
+	public void updatePersonShouldThrowMethodNotAllowedIfDoesNotExist() {
+		Person wrongPerson = new Person();
+		
+		wrongPerson.setId(WRONG_PERSON_UUID);
+		
+		when(fhirPersonService.update(WRONG_PERSON_UUID, wrongPerson)).thenThrow(MethodNotAllowedException.class);
+		
+		resourceProvider.updatePerson(new IdType().setValue(WRONG_PERSON_UUID), wrongPerson);
+	}
+	
+	@Test
+	public void deletePerson_shouldDeletePerson() {
+		Person newPerson = new Person();
```

Comment:
you can use the same person object no need to create a new one

Reply:
thanks ,i have fixed it